### PR TITLE
Standalone mode (local+clustered)

### PR DESCRIPTION
**Pull request by vblagoje**
@ryanemerson please review thoroughly. Let's make the handling and maintenance of all three modes (domain, local standalone, clustered standalone) as simple as possible.

### CODE REVIEW

**Comment by vblagoje** [comment]
@pzapataf if you have time please review as well! It is a good learning exercise!

**Comment by vblagoje** [comment]
@ryanemerson please try again as it works for me every time. Between changes of mode make sure to log out and clean cache. I made changes you requested, those were excellent suggestions. Thanks

File: src/services/jgroups/JGroupsService.ts [hunk 1]
```diff
@@ -41,84 +49,113 @@ export class JGroupsService {
...
-    };
-    return this.dmrService.readChildName(request);
+    if (this.hasJGroupsStack()) {
+      let request: IDmrRequest = <IDmrRequest>{
+        address: this.getJgroupsAddress(profile),
+        "child-type": "channel"
+      };
+      return this.dmrService.readChildName(request);
+    } else {
+      this.$q.when(["N/A"]);
```
Comment (vblagoje) [inline src/services/jgroups/JGroupsService.ts]: Right, great suggestion! Now hasJGroupsService method invocations are contained almost exclusively to JGroupsService

File: src/services/jgroups/JGroupsService.ts [hunk 2]
```diff
@@ -41,84 +49,113 @@ export class JGroupsService {
...
+        })
+        .then((stacks) => {
+          let stackMap: IMap<string> = <IMap<string>>{};
+          for (let serverGroup in stacks) {
+            stackMap[serverGroup] = stacks[serverGroup];
+          }
+          deferred.resolve(stackMap);
+        });
+      return deferred.promise;
+    }
```
Comment (vblagoje) [inline src/services/jgroups/JGroupsService.ts]: Right! Great idea!

File: src/module/server-group/ServerGroupCtrl.ts [hunk 3]
```diff
@@ -143,21 +145,31 @@ export class ServerGroupCtrl {
...
+    this.jGroupsService.getServerGroupCoordinator(this.serverGroup).then((coordinator) => {
+      this.coordinator = coordinator;
+    });
   }
 
   private fetchInetAddresses(): void {
-    this.serverGroupService.getServerInetAddresses(this.serverGroup).then((inetMap) => this.serverInetMap = inetMap);
+    this.serverGroupService.getServerInetAddresses(this.serverGroup).then((inetMap) => {
+      this.serverInetMap = inetMap;
+    });
```
Comment (vblagoje) [inline src/module/server-group/ServerGroupCtrl.ts]: Corrected!

File: src/module/cache/CacheCtrl.ts [hunk 4]
```diff
@@ -6,9 +6,11 @@ import {ICacheContainer} from "../../services/container/ICacheContainer";
 import {ICache} from "../../services/cache/ICache";
 import IModalServiceInstance = angular.ui.bootstrap.IModalServiceInstance;
 import IModalService = angular.ui.bootstrap.IModalService;
+import {JGroupsService} from "../../services/jgroups/JGroupsService";
```
Comment (ryanemerson) [inline src/module/cache/CacheCtrl.ts]: Unused import

File: src/module/navbar/NavbarCtrl.ts [hunk 5]
```diff
@@ -1,17 +1,20 @@
 import {AuthenticationService} from "../../services/authentication/AuthenticationService";
 import {IStateService} from "angular-ui-router";
 import {BRAND_NAME, BRAND_IMAGE} from "../../common/Constants";
+import {JGroupsService} from "../../services/jgroups/JGroupsService";
```
Comment (ryanemerson) [inline src/module/navbar/NavbarCtrl.ts]: Unused import

File: src/services/authentication/AuthenticationService.ts [hunk 6]
```diff
@@ -38,16 +38,25 @@ export class AuthenticationService {
...
+    this.dmrService.readResource({
+      address: [],
+      recursive: true,
+      "include-runtime": true,
+      "recursive-depth": 2
+    }).then((response: any) => {
+      let hasJGroupsStack: boolean = true;
+      let launchType: string = response["launch-type"];
+      if (LaunchTypeService.STANDALONE_MODE === launchType) {
+        hasJGroupsStack = isNotNullOrUndefined(response["subsystem"]["datagrid-jgroups"]);
```
Comment (ryanemerson) [inline src/services/authentication/AuthenticationService.ts]: let's change this to response.subsystem["datagrid-jgroups"] to keep the tslint checks happy

File: src/module/cache-container/config/ContainerConfig.ts [hunk 7]
```diff
@@ -36,8 +36,10 @@ module.config(($stateProvider: ng.ui.IStateProvider) => {
     controller: SchemaConfigCtrl,
     controllerAs: "schemaCtrl",
     resolve: {
-      availableSchemas: ["container", "schemaService", (container, schemaService) => {
-        return schemaService.getProtoSchemaNames(container);
+      availableSchemas: ["$q", "container", "schemaService", ($q:IQService, container, schemaService) => {
+        return schemaService.getProtoSchemaNames(container).catch((reason) => {
+          $q.resolve([]); // most likely standalone local mode present, simply return empty array
```
Comment (vblagoje) [inline src/module/cache-container/config/ContainerConfig.ts]: Right +1

**Comment by vblagoje** [comment]
Excellent feedback @ryanemerson Updated changes! Will do one more thorough check myself later today!

**Comment by vblagoje** [comment]
@ryanemerson are you happy with the state of this PR? If not, let me know what else needs to be addressed!

**Comment by ryanemerson** [comment]
Integrated with 76de752d0f57947fe9e91a5a8e16543e8383d0f4

**Pull request merged**